docs/topics/patterns/instant-messaging.mdx

```diff
@@ -0,0 +1,355 @@
+
+#### Sorted sets
+
+Typically used for things like [leaderboards](https://www.gomomento.com/blog/build-on-momento-tips-and-tricks-for-a-lightning-fast-game-leaderboard), a [sorted set](../../cache/develop/basics/datatypes#sorted-sets) is a collection data type that stores multiple value/score elements. In each element, the value is a `string` and the score is an `integer`. Specifically with instant messaging, you can store the time the message was sent as the *score* and a stringified JSON object as the *value*. Storing the message timestamp as the score will automatically allow you to sort messages by time since sorted sets do that by default.
+
+:::tip
+To store the time as the score, you must use an integer representation of the time. Store either ticks or time in milliseconds since the Unix epoch.
+:::
+
+Sorted sets have a [rich set of features](../../cache/develop/api-reference/sorted-set-collections) that allow you to access messages in many different ways. Consider the following use cases:
+
+* Paging messages N at a time
+* Retrieving messages in a certain time frame
+* Displaying messages in reverse order
+* Getting all messages ever sent
+
+Based on the parameters you use in your fetch command, you can easily satisfy any of the use cases above.
+
+A significant point to remember - **messages must be unique in sorted sets**! This means when saving the message to the sorted set, you should include some sort of message identifier or timestamp in your JSON.
+
+#### List
+
+A [list](../../cache/develop/basics/datatypes#lists) is a collection data type that stores an ordered array of elements. You can push elements to either the front or back of a list and you can add multiple entries at one time. A list is a great data type for instant messaging because it stores messages in the order they were sent. When users join an instant messaging session, you can quickly fetch the entire chat history with a single [listFetch API call](../../cache/develop/api-reference/list-collections#listfetch).
+
+Elements in a list do not need to be unique. You can push the same message into a list over and over again and it will be added every time. Lists are a good choice for instant messaging for the following use cases:
+
+* Paging messages N at a time
+* Adding multiple messages at once
+* Maintaining a chat history of fixed length, i.e. the last 100 messages
+
+When using a list, you have more array manipulation functions than with a sorted set, allowing you to automatically truncate message history if it gets too long or remove specific elements for things like content moderation.
+
+### Delivery
```

Comment:
Topics will facilitate real-time communication, and cache will facilitate loading historical messages for new subscribers, right? Can we call this out explicitly somewhere? It is implicit, just took me 30-60 seconds to think through. So maybe it will help others if we are explicit.